app/services/service_offering/add_to_portfolio_item.rb

```diff
@@ -0,0 +1,50 @@
+module ServiceOffering
+  class AddToPortfolioItem
+    IGNORE_FIELDS = %w(id created_at updated_at portfolio_id).freeze
+
+    attr_reader :params
+
+    def initialize(params)
+      @params = HashWithIndifferentAccess.new(params)
+    end
+
+    def process
+      # Get the api response object from Topology
+      TopologicalInventory.call do |api_instance|
+        @service_offering = api_instance.show_service_offering(@params[:service_offering_ref])
+      end
+
+      # Get the fields that we're going to pull over
+      create_param_map(determine_valid_fields)
+
+      PortfolioItem.create!(populate_missing_fields)
```

Comment:
@lindgrenj6 @syncrou 
Based on my discussion with @Fryguy the process function should return the service object and then the controller can decide which response object needs to send back. In this case I am thinking we would store the  result of PortfolioItem.create into an instance variable that can be accessed from the service object

Reply:
I refactored it to store the `PortfolioItem` into an instance var called `@item`, and made it return self. The controller then accesses the `item` instance variable to render.